From eb11fcca9c37f18d8e5f3d92aa4dddf4c0ca9a8c Mon Sep 17 00:00:00 2001
From: Qin Su <qsu@ti.com>
Date: Tue, 27 Nov 2018 18:15:49 -0500
Subject: [PATCH] enable use of arm compute shared library

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Qin Su <qsu@ti.com>

---
 cmake/GlobalConfig.cmake | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cmake/GlobalConfig.cmake b/cmake/GlobalConfig.cmake
index 12a271af..d98ff111 100755
--- a/cmake/GlobalConfig.cmake
+++ b/cmake/GlobalConfig.cmake
@@ -280,6 +280,11 @@ if(ARMCOMPUTENEON OR ARMCOMPUTECL)
         find_library(ARMCOMPUTE_CORE_LIBRARY_DEBUG NAMES arm_compute_core-static)
         find_library(ARMCOMPUTE_CORE_LIBRARY_RELEASE NAMES arm_compute_core-static)
 
+        find_library(ARMCOMPUTE_LIBRARY_DEBUG NAMES arm_compute)
+        find_library(ARMCOMPUTE_LIBRARY_RELEASE NAMES arm_compute)
+        find_library(ARMCOMPUTE_CORE_LIBRARY_DEBUG NAMES arm_compute_core)
+        find_library(ARMCOMPUTE_CORE_LIBRARY_RELEASE NAMES arm_compute_core)
+
         set(ARMCOMPUTE_LIBRARIES
             debug ${ARMCOMPUTE_LIBRARY_DEBUG} ${ARMCOMPUTE_CORE_LIBRARY_DEBUG}
             optimized ${ARMCOMPUTE_LIBRARY_RELEASE} ${ARMCOMPUTE_CORE_LIBRARY_RELEASE} )
